sql-plugin/src/main/scala/org/apache/spark/sql/rapids/AggregateFunctions.scala

```diff
@@ -891,3 +896,56 @@ case class GpuCollectSet(
       inputs: Seq[(ColumnVector, Int)]): RollingAggregationOnColumn =
     RollingAggregation.collectSet().onColumn(inputs.head._2)
 }
+
+trait CpuToGpuAggregateBufferConverter {
+  def createExpression(child: Expression): CpuToGpuBufferTransition
+}
+
+trait GpuToCpuAggregateBufferConverter {
+  def createExpression(child: Expression): GpuToCpuBufferTransition
+}
+
+trait CpuToGpuBufferTransition extends UnaryExpression with CodegenFallback
+
+trait GpuToCpuBufferTransition extends UnaryExpression with CodegenFallback {
+  override def dataType: DataType = BinaryType
+}
+
+class CpuToGpuCollectBufferConverter(
+    elementType: DataType) extends CpuToGpuAggregateBufferConverter {
+  def createExpression(child: Expression): CpuToGpuBufferTransition = {
+    CpuToGpuCollectBufferTransition(child, elementType)
+  }
+}
+
+case class CpuToGpuCollectBufferTransition(
+    override val child: Expression,
+    private val elementType: DataType) extends CpuToGpuBufferTransition {
+
+  private lazy val row = new UnsafeRow(1)
+
+  override def dataType: DataType = ArrayType(elementType, containsNull = false)
+
+  override protected def nullSafeEval(input: Any): ArrayData = {
+    val bytes = input.asInstanceOf[Array[Byte]]
```

Comment:
Some comments here might be good to explain what is happening and point to the code that Spark uses to serialize out the data.